src/app/core/services/rundown-timing.service.ts

```diff
@@ -5,6 +5,11 @@ import { PartEntityService } from './models/part-entity.service'
 import { Part } from '../models/part'
 import { Injectable } from '@angular/core'
 
+interface AccumulatedResult<Result, Accumulator> {
+  result: Result
+  accumulator: Accumulator
+}
```

Comment:
This is a quite confusing interface. I have been looking at the places you use it and I can't tell what it is the values for `result` and `accumulator` is supposed to mean.

Reply:
Only because i have seen the tests for `getStartOffsetsInMsFromNextCursorForSegments` do i have an Okay understanding of what this interface does.

That said, without the tests showing an example of how it is used in pratice, I too would have been very confused.

Maybe renaming `accumulator` to something along the lines of `accumulatedMsOffset` could help. What ever renaming is applied should also be appled similarly in `getStartOffsetsInMsFromNextCursorForSegments`.

Reply:
I have specialized the interface to the following snippet and have tried to simplify the method that uses it.
```typescript
interface AccumulatedStartOffsetInMsForSegmentsData {
  startOffsetsInMsFromNextCursorForSegments: Record<string, number>
  accumulatedStartOffsetInMsFromNextCursor: number
}
```

Reply:
Hmm, in my mind the longer names / extraction to a seperate method didn't help make it more clear what happens.

I have an idea for a possibility refactor, that i'd like to spend a bit of time on, on monday, if it is alright with you. I will be branching from your branch, and then creaing a small seperate PR, so you can review my idea.

Or maybe you just want to talk me throught the idea with you. Let me know what you prefer.

Reply:
The more specific names helped :)